### PR TITLE
Add documentation on how to debug the SageMaker orchestrator

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
See title.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md [hunk 1]
```diff
@@ -119,6 +133,27 @@ side. From there you can view the pipelines that have been launched via ZenML:
...
+its only execution, then click on the failed step in the pipeline graph and go
+to the 'Output' tab:
+
+![SageMaker Studio Logs](../../../.gitbook/assets/sagemaker-logs.png)
+
+Alternatively, you can use 
+[Amazon CloudWatch](https://aws.amazon.com/cloudwatch/) to get an even more 
+detailed view of all the log messages that were written during the execution of 
+your SageMaker pipeline. To do so, search for 'CloudWatch' in the search bar of
+your AWS terminal, navigate to 'Logs > Log groups' and open the
```
Comment (fa9r) [inline docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:151]:
```suggestion
your AWS console, navigate to 'Logs > Log groups' and open the
```

File: docs/book/.gitbook/assets/sagemaker-cloudwatch-logs.png
Comment (strickvl) [inline docs/book/.gitbook/assets/sagemaker-cloudwatch-logs.png:1]: Not sure, but if one were being paranoid, maybe worth scrubbing / removing the role name in the upper right corner?

File: docs/book/.gitbook/assets/sagemaker-logs.png
Comment (strickvl) [inline docs/book/.gitbook/assets/sagemaker-logs.png:1]: Maybe same here for the referenced role?

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md [hunk 2]
```diff
@@ -119,6 +133,27 @@ side. From there you can view the pipelines that have been launched via ZenML:
 
 ![Sagemaker Studio Pipelines](../../../.gitbook/assets/sagemakerUI.png)
 
+### Debugging SageMaker Pipelines
+
+If your SageMaker pipeline failed before the first ZenML step was started, the
+ZenML run will not show up in the ZenML dashboard. In these cases, you need to 
+[use the SageMaker UI](#sagemaker-ui) to view the respective error message and
+logs. To do so, open the corresponding pipeline in the SageMaker UI, then open
```
Comment (strickvl) [inline docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:141]: A lot of assumption that the person will know how to open the pipeline in the Sagemaker UI. Maybe just a bit more information / hints about where they can go to find this?

**Review by strickvl — CHANGES_REQUESTED** [review]

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md (hunk 2 above)
Comment (fa9r) [reply docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:141]: That's shown in a lot of detail incl. screenshots in the previous section (#sagemaker-ui)

**Review by strickvl — APPROVED** [review]

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md [hunk 3]
```diff
@@ -119,6 +133,27 @@ side. From there you can view the pipelines that have been launched via ZenML:
...
+its only execution, then click on the failed step in the pipeline graph and go
+to the 'Output' tab:
```
Comment (safoinme) [inline docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:143]: I personally find bullet points or numbered lists a better approach to describe series of steps but this paragraph is also great.
```suggestion
If your SageMaker pipeline encounters an error before the first ZenML step starts, the ZenML run will not appear in the ZenML dashboard. In such cases, use the [SageMaker UI](#sagemaker-ui) to review the error message and logs. Here's how:

Open the corresponding pipeline in the SageMaker UI.
Open its execution.
Click on the failed step in the pipeline graph.
Go to the 'Output' tab to access logs.
```

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md [hunk 4]
```diff
@@ -119,6 +133,27 @@ side. From there you can view the pipelines that have been launched via ZenML:
...
+
+![SageMaker Studio Logs](../../../.gitbook/assets/sagemaker-logs.png)
+
+Alternatively, you can use 
+[Amazon CloudWatch](https://aws.amazon.com/cloudwatch/) to get an even more 
+detailed view of all the log messages that were written during the execution of 
+your SageMaker pipeline. To do so, search for 'CloudWatch' in the search bar of
+your AWS console, navigate to 'Logs > Log groups' and open the
+'/aws/sagemaker/ProcessingJobs' log group. Here you will be able to see one log
+stream per step for each of your SageMaker pipeline executions:
```
Comment (safoinme) [inline docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:153]:
```suggestion
Alternatively, for a more detailed view of log messages during SageMaker pipeline execution, consider using [Amazon CloudWatch](https://aws.amazon.com/cloudwatch/):

Search for 'CloudWatch' in the AWS console search bar.
Navigate to 'Logs > Log groups.'
Open the '/aws/sagemaker/ProcessingJobs' log group.
Here, you can find log streams for each step in your SageMaker pipeline executions.
```

**Review by safoinme — APPROVED** [review]
Just a minor comment around structure of the paragraph (can be ignored)

File: docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md (hunk 3 above)
Comment (fa9r) [reply docs/book/stacks-and-components/component-guide/orchestrators/sagemaker.md:143]: Great suggestions 👍

**Pull request merged**